### ISSUE TITLE
#1383 - Guest UI: "Existing Reservation" option and "Event Horizon" violations displaying at inappropriate times.

### ISSUE THREAD

**Issue by NishantUpadhyay-BTC** [NishantUpadhyay-BTC/BLISS-Issue-Tracking#318]
Sometimes a user is not logged in, is starting a new reservation, and is presented with the option for “Return to Existing Reservation" or an event horizon violation notice which doesn't apply to the dates currently chosen, but would apply to a previous search made during an earlier session when that user was logged in.

I've seen this happen three times in the last week, but cannot duplicate it. I think it might be caused by stale data cached in the browser from running several test reservations from the same computer - logging in/out, etc, and that data maybe not always clearing out as as it should. I have no proof of this of course, but it's the only possible cause I could come up with - or are we cacheing user IP info and reservation state server side? Maybe that isn't clearing properly across multiple log in/log out and test reservations?

 a clue here. I just returned to testing after a few hours break. I was on the "Personal Retreat Availability" page, having gotten there when by hitting the 'Cancel and Lo" on the Guest Details page a few hours before. I noticed that the progress bar looked odd. I was on the "Basic Details part of the reservation, but "Reservation Details" was also dark gray - indicating I'd already visited that portion of the reservation. Now that should have been cleared out when I hit "Cancel and Return to Availability". I proceeded with starting a new reservation, found a lodging option and hit "reserve now"- and sure enough I got the message asking me if I wanted to return to my previous reservation or start a new one. I should not have gotten this message, since I chose to cancel it earlier - but based on what is being displayed in the Progress Bar, I think there is _something_ being cached client-side that makes the App think I still have a reservation in progress. At that point I chose to "Create a New Reservation". I was taken to the login page, and then, surprisingly after I logged in I saw a warning message that said I previously had a reservation in progress but it had timed out - please start over. And I was dropped back onto the PR Availability page instead of being taken to the Guest Details page of the reservation I'd just begun.

So - something is definitely screwy here.

**Comment by neerajkhatri-btc** [comment]
@NishantUpadhyay-BTC : this issue is fixed.

**Comment by NishantUpadhyay-BTC** [comment]
Today I went to the site for testing after not having even visited the site for several days. I ran a lodging search per usual and when I clicked "reserve now" was presented with the dialog box that asked me if I wanted to continue with my existing reservation or start a new one. Why? I wasn't even logged in yet. This should not be possible for a user that is not even logged in, but even if I had logged in, there should be no pending reservations, as it's been days since I've even been in the system.
